File: app/src/main/java/ru/magflayer/spectrum/domain/repository/ColorInfoRepository.java
package ru.magflayer.spectrum.domain.repository;

import java.util.List;
import java.util.Map;

import ru.magflayer.spectrum.domain.entity.ColorInfoEntity;

public interface ColorInfoRepository {

    boolean uploadColorNames(Map<String, String> hexName);

    boolean isColorNamesUploaded();

    List<ColorInfoEntity> loadColorNames();

    String loadColorNameByHex(String hex);

    boolean uploadNcsColors(Map<String, String> hexName);

    boolean isNcsColorUploaded();

    List<ColorInfoEntity> loadNcsColors();

    String loadNcsColorByHex(String hex);

}

File: app/src/main/java/ru/magflayer/spectrum/domain/repository/ColorInfoRepository.kt
package ru.magflayer.spectrum.domain.repository

import ru.magflayer.spectrum.domain.entity.ColorInfoEntity

interface ColorInfoRepository {

    suspend fun isColorNamesUploaded(): Boolean

    suspend fun isNcsColorUploaded(): Boolean

    suspend fun uploadColorNames(hexName: Map<String, String>): Boolean

    suspend fun loadColorNames(): List<ColorInfoEntity>

    suspend fun loadColorNameByHex(hex: String): String

    suspend fun uploadNcsColors(hexName: Map<String, String>): Boolean

    suspend fun loadNcsColors(): List<ColorInfoEntity>

    suspend fun loadNcsColorByHex(hex: String): String
}
